File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums) # O(n)
        longest_sequence = 0

        for num in nums_set: # O(n)
            if (num - 1) not in nums_set:
                current_num = num
                current_sequence = 1
            
                while current_num + 1 in nums_set: # O(1) 
                    current_num += 1
                    current_sequence += 1

                longest_sequence = max(current_sequence, longest_sequence)

        return longest_sequence

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> uniqueNums = Arrays.stream(nums).boxed().collect(Collectors.toSet());
        int result = 0;

        for (int num : nums) {
            if (uniqueNums.contains(num - 1)) continue;
            int length = 1;
            while (uniqueNums.contains(num + length)) length += 1;
            result = Math.max(length, result);
        }

        return result;
    }
}